File: packages/navigation/android/src/main/java/expo/modules/gaodemap/map/overlays/MarkerViewModule.kt
package expo.modules.gaodemap.map.overlays

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/**
 * Marker 视图 Module
 */
class MarkerViewModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("MarkerView")

    View(MarkerView::class) {
      Events("onMarkerPress", "onMarkerDragStart", "onMarkerDrag", "onMarkerDragEnd")
      
      // 纬度
      Prop<Double>("latitude") { view, lat ->
        view.setLatitude(lat)
      }
      // 经度
      Prop<Double>("longitude") { view, lng ->
        view.setLongitude(lng)
      }
      // 位置
      Prop<Map<String, Any>?>("position") { view, position ->
        view.setPosition(position)
      }
      // 标题
      Prop<String>("title") { view, title ->
        view.setTitle(title)
      }
      // 副标题
      Prop<String>("snippet") { view, snippet ->
        view.setDescription(snippet)
      }
      // 是否可拖拽
      Prop<Boolean>("draggable") { view, draggable ->
        view.setDraggable(draggable)
      }
      // 图标
      Prop<String>("icon") { view, icon ->
        view.setMarkerIcon(icon)
      }
      // 图标颜色
      Prop<String>("pinColor") { view, color ->
        view.setPinColor(color)
      }
      // 透明度
      Prop<Float>("opacity") { view, opacity ->
        view.setOpacity(opacity)
      }
      // 是否平铺
      Prop<Boolean>("flat") { view, flat ->
        view.setFlat(flat)
      }
      // 层级
      Prop<Float>("zIndex") { view, zIndex ->
        view.setZIndex(zIndex)
      }
      // 锚点
      Prop<Map<String, Float>>("anchor") { view, anchor ->
        view.setAnchor(anchor)
      }
      // 图标宽度
      Prop<Int>("iconWidth") { view, width ->
        view.setIconWidth(width)
      }
      // 图标高度
      Prop<Int>("iconHeight") { view, height ->
        view.setIconHeight(height)
      }
      // 自定义视图宽度
      Prop<Int>("customViewWidth") { view, width ->
        view.setCustomViewWidth(width)
      }
      // 自定义视图高度
      Prop<Int>("customViewHeight") { view, height ->
        view.setCustomViewHeight(height)
      }
        // 缓存key
      Prop<String>("cacheKey") { view, key ->
            view.setCacheKey(key)
      }

      // 平滑移动路径
      Prop<List<Any>?>("smoothMovePath") { view: MarkerView, path ->
        view.setSmoothMovePath(path)
      }

      // 平滑移动时长（秒）
      Prop<Double>("smoothMoveDuration") { view, duration ->
        view.setSmoothMoveDuration(duration)
      }

      // 生长动画
      Prop<Boolean>("growAnimation") { view, enable ->
        view.setGrowAnimation(enable)
      }
    }
  }
}

File: packages/core/ios/overlays/MarkerViewModule.swift
import ExpoModulesCore

public class MarkerViewModule: Module {
    public func definition() -> ModuleDefinition {
        Name("MarkerView")
        
        View(MarkerView.self) {
            // 🔑 声明专属事件（避免与其他组件冲突）
            Events("onMarkerPress", "onMarkerDragStart", "onMarkerDrag", "onMarkerDragEnd")
            
            // 拆分 position 为两个独立属性以兼容 React Native 旧架构
            Prop("latitude") { (view: MarkerView, lat: Double) in
                view.setLatitude(lat)
            }
            
            Prop("longitude") { (view: MarkerView, lng: Double) in
                view.setLongitude(lng)
            }
            
            Prop("title") { (view: MarkerView, title: String) in
                view.setTitle(title)
            }
            
            Prop("snippet") { (view: MarkerView, snippet: String) in
                view.setDescription(snippet)
            }
            
            Prop("draggable") { (view: MarkerView, draggable: Bool) in
                view.setDraggable(draggable)
            }
            
            Prop("icon") { (view: MarkerView, source: String?) in
                view.setIconUri(source)
            }
            
            Prop("iconWidth") { (view: MarkerView, width: Double) in
                view.setIconWidth(width)
            }
            
            Prop("iconHeight") { (view: MarkerView, height: Double) in
                view.setIconHeight(height)
            }
            
            Prop("customViewWidth") { (view: MarkerView, width: Double) in
                view.setCustomViewWidth(width)
            }
            
            Prop("customViewHeight") { (view: MarkerView, height: Double) in
                view.setCustomViewHeight(height)
            }
            
            Prop("centerOffset") { (view: MarkerView, offset: [String: Double]) in
                view.setCenterOffset(offset)
            }
            
            Prop("animatesDrop") { (view: MarkerView, animate: Bool) in
                view.setAnimatesDrop(animate)
            }
            
            Prop("pinColor") { (view: MarkerView, color: String) in
                view.setPinColor(color)
            }
            
            Prop("canShowCallout") { (view: MarkerView, canShow: Bool) in
                view.setCanShowCallout(canShow)
            }

            Prop("growAnimation") { (view: MarkerView, enabled: Bool) in
                view.setGrowAnimation(enabled)
            }
             Prop("cacheKey") { (view: MarkerView, key: String) in
                view.setCacheKey(key)
            }
            
            Prop("position") { (view: MarkerView, position: [String: Double]?) in
                view.setPosition(position)
            }
            
            // 平滑移动路径
            Prop("smoothMovePath") { (view: MarkerView, path: [[String: Double]]) in
                view.setSmoothMovePath(path)
            }
            
            // 平滑移动时长（秒）
            Prop("smoothMoveDuration") { (view: MarkerView, duration: Double) in
                view.setSmoothMoveDuration(duration)
            }
        }
    }
}
